### PR TITLE
Update dev dependencies

### LINKED ISSUE #1451: Upgrade all deps and dev deps that could be easily upgraded
### Description

Some of the dependencies should be possible to upgrade easily with no additional effort. Let's do that.

### Links
- https://handsoncode.slack.com/archives/D02UEUFJWHZ/p1730151412323929 (accessible by @sequba)

### PR DESCRIPTION
**Pull request by sequba**
### Current HF dependencies:

dependencies:
    chevrotain: ⚠️ next version contains significant breaking changes, more work needed
    tiny-emitter: ✅ already latest version

devDependencies:
    @babel/cli: ✅ upgraded to latest version
    @babel/core: ✅ upgraded to latest version
    @babel/plugin-transform-runtime: ✅ upgraded to latest version
    @babel/preset-env: ✅ upgraded to latest version
    @babel/register: ✅ upgraded to latest version
    @babel/runtime: ✅ upgraded to latest version
    @babel/preset-react: ✅ upgraded to latest version
    @babel/preset-typescript: ✅ upgraded to latest version
    @babel/plugin-transform-modules-commonjs: ✅ upgraded to latest version
    @babel/plugin-syntax-class-properties: ✅ already latest version
    @babel/plugin-proposal-decorators: ✅ upgraded to latest version
    @babel/plugin-proposal-class-properties: ✅ upgraded to latest version
    @babel/plugin-proposal-private-methods: ✅ upgraded to latest version
    @babel/plugin-proposal-private-property-in-object: ✅ upgraded to latest version
    @babel/plugin-transform-typescript: 🗑️ removed, not needed anymore
    @microsoft/tsdoc: 🗑️ removed, not needed anymore
    @types/jasmine: ✅ upgraded to latest version
    @types/jest: ⚠️ upgraded, but next version requires updating typescript
    @types/jsdom: ✅ upgraded to latest version
    @types/node: ⚠️ upgraded, but next version contains significant breaking changes, more work needed
    @types/webpack-env: ✅ upgraded to latest version
    @typescript-eslint/eslint-plugin: ⚠️ upgraded, but next version requires updating eslint
    @typescript-eslint/parser: ⚠️ upgraded, but next version requires updating eslint
    @vuepress/plugin-active-header-links: 🗑️ removed, not needed anymore
    as-table: ✅ already latest version
    babel-loader: ⚠️ upgraded, but next version requires updating webpack
    babel-plugin-transform-inline-environment-variables: ✅ upgraded to latest version
    core-js: ✅ upgraded to latest version
    cross-env: ✅ upgraded to latest version
    env-cmd: ✅ already latest version
    eslint: ⚠️ upgraded, but next version contains significant breaking changes, more work needed
    eslint-config-prettier: "✅ upgraded to latest version
    eslint-plugin-jsdoc: ✅ upgraded to latest version
    eslint-plugin-license-header: ✅ upgraded to latest version
    eslint-plugin-prettier: ✅ upgraded to latest version
    full-icu: ✅ upgraded to latest version
    jasmine: ✅ upgraded to latest version
    jest: ⚠️ upgraded, but next version requires updating typescript
    jsdom: ✅ upgraded to latest version
    karma: ✅ upgraded to latest version
    karma-chrome-launcher: ✅ upgraded to latest version
    karma-firefox-launcher: ✅ upgraded to latest version
    karma-jasmine: ✅ upgraded to latest version
    karma-jasmine-html-reporter: ✅ upgraded to latest version
    karma-sourcemap-loader: ✅ upgraded to latest version
    karma-webpack: ⚠️next version requires updating webpack
    license-checker: ✅ already latest version
    markdown-it-footnote: ✅ upgraded to latest version
    markdown-it-regex: ✅ already latest version
    moment: ✅ upgraded to latest version
    npm-run-all: ✅ already latest version
    on-build-webpack: ✅ already latest version
    rimraf: ✅ upgraded to latest version
    serve: 🗑️ removed, not needed anymore
    string-replace-loader: ⚠️next version requires updating webpack
    tar: ✅ upgraded to latest version
    terser-webpack-plugin: ⚠️next version requires updating webpack
    ts-jest: ⚠️ upgraded, but next version requires updating typescript
    ts-loader: ⚠️upgraded, but next version requires updating webpack
    ts-node: ✅ upgraded to latest version
    typedoc: ⚠️ next version contains significant breaking changes, more work needed
    typedoc-plugin-markdown: ⚠️ upgraded, but next version requires updating typedoc
    typescript: ⚠️ next version contains significant breaking changes, more work needed
    vuepress: ⚠️upgraded, but next version contains significant breaking changes, more work needed
    vuepress-plugin-clean-urls: 🗑️ removed, not needed anymore
    weak-napi: 🗑️ removed, not needed anymore
    webpack: ⚠️ upgraded, but next version contains significant breaking changes, more work needed
    webpack-cli: ⚠️ upgraded, but next version requires updating webpack
    webpackbar: ⚠️ upgraded, but next version requires updating webpack

### How did you test your changes?

- [x] linter
- [x] unit tests
- [x] unit tests (browser)
- [x] local package build
- [x] local docs build

### Types of changes

- [x] Update of the dev dependencies

### Related issues:
Fixes #1451

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [ ] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,85 +93,77 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.62.0",
+    "@typescript-eslint/parser": "^5.62.0",
     "as-table": "^1.0.55",
-    "babel-loader": "^8.0.6",
-    "babel-plugin-transform-inline-environment-variables": "^0.4.3",
-    "core-js": "^3.23.0",
-    "cross-env": "^7.0.0",
+    "babel-loader": "^8.4.1",
+    "babel-plugin-transform-inline-environment-variables": "^0.4.4",
+    "core-js": "^3.39.0",
```
Comment (budnix) [inline package.json:128]: After bumping the `core-js,`, you should also update the version in [babel config](https://github.com/handsontable/hyperformula/blob/feature/issue-1451/babel.config.js#L12).

File: package.json [hunk 2]
```diff
@@ -95,85 +93,77 @@
...
-    "@babel/plugin-transform-runtime": "^7.8.3",
-    "@babel/preset-env": "^7.8.4",
-    "@babel/register": "^7.9.0",
-    "@babel/runtime": "^7.18.9",
-    "@babel/preset-react": "^7.13.13",
-    "@babel/preset-typescript": "^7.13.0",
-    "@babel/plugin-transform-modules-commonjs": "^7.13.8",
+    "@babel/cli": "^7.25.9",
+    "@babel/core": "^7.26.0",
+    "@babel/plugin-transform-runtime": "^7.25.9",
```
Comment (budnix) [inline package.json:106]: The same as for `core-js`? https://github.com/handsontable/hyperformula/blob/feature/issue-1451/babel.config.js#L21

**Review by budnix — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sequba) [reply package.json:128]: Done. Thank you for spotting it ;)

File: package.json (hunk 2 above)
Comment (sequba) [reply package.json:106]: Done. Thank you for spotting it ;)

**Review by budnix — APPROVED** [review]

**Pull request merged**